File: LegendDataEditor/LegendDataEditor/Sqlite/Table/SQLiteMonster.cpp
#include "SQLiteMonster.h"
#include "Utility.h"
#include "SQLiteDataReader.h"
#include "SQLite.h"

string MonsterData::COL_ID = "ID";
string MonsterData::COL_DESCRIBE = "Describe";
string MonsterData::COL_TEMPLATE_ID = "TemplateID";
string MonsterData::COL_LEVEL = "Level";
string MonsterData::COL_HP = "HP";
string MonsterData::COL_MIN_PHYSICS_DEFENCE = "MinPhysicsDefence";
string MonsterData::COL_MAX_PHYSICS_DEFENCE = "MaxPhysicsDefence";
string MonsterData::COL_MIN_MAGIC_DEFENCE = "MinMagicDefence";
string MonsterData::COL_MAX_MAGIC_DEFENCE = "MaxMagicDefence";
string MonsterData::COL_MIN_PHYSICS_ATTACK = "MinPhysicsAttack";
string MonsterData::COL_MAX_PHYSICS_ATTACK = "MaxPhysicsAttack";
string MonsterData::COL_MIN_MAGIC_ATTACK = "MinMagicAttack";
string MonsterData::COL_MAX_MAGIC_ATTACK = "MaxMagicAttack";
string MonsterData::COL_MIN_SORCERY_ATTACK = "MinSorceryAttack";
string MonsterData::COL_MAX_SORCERY_ATTACK = "MaxSorceryAttack";
string MonsterData::COL_MOVE_SPEED = "MoveSpeed";
string MonsterData::COL_HIT_PROBABLITY = "HitProbability";
string MonsterData::COL_EVADE_PROBABLITY = "EvadeProbability";
string MonsterData::COL_WALK_SPEED = "WalkSpeed";
string MonsterData::COL_WALK_STEP = "WalkStep";
string MonsterData::COL_WALK_WAIT = "WalkWait";
string MonsterData::COL_ATTACK_SPEED = "AttackSpeed";
string MonsterData::COL_EXP = "Exp";
string MonsterData::COL_DROP_LIST = "DropList";
string MonsterData::COL_SKILL = "Skill";
string MonsterData::COL_SKILL_LEVEL = "SkillLevel";

void SQLiteMonster::query(int id, MonsterData& data)
{
	string conditionString;
	StringUtility::appendConditionInt(conditionString, MonsterData::COL_ID, id, "");
	doSelect(data, conditionString);
}

bool SQLiteMonster::insert(const MonsterData& data)
{
	string valueString;
	data.insert(valueString);
	StringUtility::removeLastComma(valueString);
	return doInsert(valueString);
}

bool SQLiteMonster::update(const MonsterData& data)
{
	string updateString;
	data.update(updateString);
	StringUtility::removeLastComma(updateString);
	string conditionStr;
	StringUtility::appendConditionInt(conditionStr, MonsterData::COL_ID, data.mID, "");
	return doUpdate(updateString, conditionStr);
}

File: LegendEditor/Assets/Scripts/Game/DataBase/SQLiteTable/SQLiteMonster.cs
using Mono.Data.Sqlite;
using System;
using System.Collections.Generic;

public class TDMonster : TableData
{
	public static string Describe = "Describe";
	public static string TemplateID = "TemplateID";
	public static string Level = "Level";
	public static string HP = "HP";
	public static string MinPhysicsDefence = "MinPhysicsDefence";
	public static string MaxPhysicsDefence = "MaxPhysicsDefence";
	public static string MinMagicDefence = "MinMagicDefence";
	public static string MaxMagicDefence = "MaxMagicDefence";
	public static string MinPhysicsAttack = "MinPhysicsAttack";
	public static string MaxPhysicsAttack = "MaxPhysicsAttack";
	public static string MinMagicAttack = "MinMagicAttack";
	public static string MaxMagicAttack = "MaxMagicAttack";
	public static string MinSorceryAttack = "MinSorceryAttack";
	public static string MaxSorceryAttack = "MaxSorceryAttack";
	public static string MoveSpeed = "MoveSpeed";
	public static string HitProbability = "HitProbability";
	public static string EvadeProbability = "EvadeProbability";
	public static string WalkSpeed = "WalkSpeed";
	public static string WalkStep = "WalkStep";
	public static string WalkWait = "WalkWait";
	public static string AttackSpeed = "AttackSpeed";
	public static string Exp = "Exp";
	public static string DropList = "DropList";
	public static string Skill = "Skill";
	public static string SkillLevel = "SkillLevel";
	public static string SearchTargetDistance = "SearchTargetDistance";
	public static string TelentBuff = "TelentBuff";
	public string mDescribe;
	public int mTemplateID;
	public int mLevel;
	public int mHP;
	public int mMinPhysicsDefence;
	public int mMaxPhysicsDefence;
	public int mMinMagicDefence;
	public int mMaxMagicDefence;
	public int mMinPhysicsAttack;
	public int mMaxPhysicsAttack;
	public int mMinMagicAttack;
	public int mMaxMagicAttack;
	public int mMinSorceryAttack;
	public int mMaxSorceryAttack;
	public float mMoveSpeed;
	public float mHitProbability;
	public float mEvadeProbability;
	public int mWalkSpeed;
	public float mWalkStep;            // 一次移动的距离,单位像素
	public float mWalkWait;            // 移动每步的等待,单位秒
	public int mAttackSpeed;
	public int mExp;
	public string mDropList;
	public List<int> mSkill = new List<int>();
	public List<int> mSkillLevel = new List<int>();
	public int mSearchTargetDistance;
	public List<int> mTelentBuff = new List<int>();
	public override void parse(SqliteDataReader reader)
	{
		base.parse(reader);
		parseParam(reader, ref mDescribe, Describe);
		parseParam(reader, ref mTemplateID, TemplateID);
		parseParam(reader, ref mLevel, Level);
		parseParam(reader, ref mHP, HP);
		parseParam(reader, ref mMinPhysicsDefence, MinPhysicsDefence);
		parseParam(reader, ref mMaxPhysicsDefence, MaxPhysicsDefence);
		parseParam(reader, ref mMinMagicDefence, MinMagicDefence);
		parseParam(reader, ref mMaxMagicDefence, MaxMagicDefence);
		parseParam(reader, ref mMinPhysicsAttack, MinPhysicsAttack);
		parseParam(reader, ref mMaxPhysicsAttack, MaxPhysicsAttack);
		parseParam(reader, ref mMinMagicAttack, MinMagicAttack);
		parseParam(reader, ref mMaxMagicAttack, MaxMagicAttack);
		parseParam(reader, ref mMinSorceryAttack, MinSorceryAttack);
		parseParam(reader, ref mMaxSorceryAttack, MaxSorceryAttack);
		parseParam(reader, ref mMoveSpeed, MoveSpeed);
		parseParam(reader, ref mHitProbability, HitProbability);
		parseParam(reader, ref mEvadeProbability, EvadeProbability);
		parseParam(reader, ref mWalkSpeed, WalkSpeed);
		parseParam(reader, ref mWalkStep, WalkStep);
		parseParam(reader, ref mWalkWait, WalkWait);
		parseParam(reader, ref mAttackSpeed, AttackSpeed);
		parseParam(reader, ref mExp, Exp);
		parseParam(reader, ref mDropList, DropList);
		parseParam(reader, ref mSkill, Skill);
		parseParam(reader, ref mSkillLevel, SkillLevel);
		parseParam(reader, ref mSearchTargetDistance, SearchTargetDistance);
		parseParam(reader, ref mTelentBuff, TelentBuff);
	}
}

public class SQLiteMonster : SQLiteTable
{
	public SQLiteMonster()
		:base(typeof(TDMonster)){}
}